tests/Rule/RuleTest.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Sabberworm\CSS\Tests\Value;
+
+use PHPUnit\Framework\TestCase;
+use Sabberworm\CSS\Parsing\ParserState;
+use Sabberworm\CSS\Settings;
+use Sabberworm\CSS\Rule\Rule;
+use Sabberworm\CSS\Value\RuleValueList;
+use Sabberworm\CSS\Value\ValueList;
+
+/**
+ * @covers \Sabberworm\CSS\Rule\Rule
+ */
+final class RuleTest extends TestCase
+{
+    /**
+     * @return array<string, array{0: string, 1: array<int, string>}>
+     */
+    public static function provideRulesAndExpectedParsedValueListTypes(): array
+    {
+        return [
+            'src (e.g. in @font-face)' => [
+                "
+                    src: url('../fonts/open-sans-italic-300.woff2') format('woff2'),
+                         url('../fonts/open-sans-italic-300.ttf') format('truetype');
+                ",
+                [RuleValueList::class, RuleValueList::class],
+            ],
+        ];
+    }
+
+    /**
+     * @test
+     *
+     * @param array<int, string> $expectedTypeClassnames
+     *
+     * @dataProvider provideRulesAndExpectedParsedValueListTypes
+     */
+    public function parsesValuesIntoExpectedTypeList(string $rule, array $expectedTypeClassnames): void
+    {
+        $subject = Rule::parse(new ParserState($rule, Settings::create()));
+
+        $value = $subject->getValue();
+        self::assertInstanceOf(ValueList::class, $value);
+
+        $actualClassnames = \array_map(
+            function ($component): string {
+                return \get_class($component);
+            },
+            $value->getListComponents()
+        );
+
+        self::assertSame($expectedTypeClassnames, $actualClassnames);
```

Comment:
Could we maybe use [`assertContainsOnlyInstancesOf`](https://docs.phpunit.de/en/8.5/assertions.html#assertcontainsonlyinstancesof) instead and avoid the anonymous function?

Reply:
It's true that the only test data added so far does on;y contain `RuleValueList`s.  But other properties may be expected to be parsed into different `Value`-based classes.